File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
imput = sys.stdin.readline

n = int(input())
n_list = list( map(int,input().split()))
n_list.sort()

m = int(input())
m_list = list( map(int,input().split()))

def find_number(seq,n):
    left = 0
    right = len(seq) - 1
    while left <= right:
        mid = (left + right) // 2
        if seq[mid] == n:
            return 1
        elif seq[mid] < n:
            left = mid + 1
        else:
            right = mid - 1
    return 0

for i in m_list:
    print(find_number(n_list,i))

File: 백준/Silver/1920. 수 찾기/수 찾기.js
const input = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');

const [_1, arr1, _2, arr2] = input;
const list1 = arr1
    .split(' ')
    .map((item) => Number(item))
    .sort((a, b) => a - b);
const list2 = arr2.split(' ').map((item) => Number(item));

const binarySearch = (arr, target) => {
    let start = 0;
    let end = arr.length - 1;

    while (start <= end) {
        const mid = Math.floor((start + end) / 2);
        if (arr[mid] === target) return mid;
        if (arr[mid] > target) end = mid - 1;
        else start = mid + 1;
    }
    return -1;
};

list2.forEach((item) => {
    console.log(binarySearch(list1, item) !== -1 ? 1 : 0);
});
